test/integration/user-management-api/service_contracts_controller_test.rb

```diff
@@ -25,6 +30,44 @@ def test_index
       assert xml.xpath('.//service_contracts/service-contract/id').text == @service_contract.id.to_s
     end
 
+    def test_show # to get a service contract
+      get admin_api_account_service_contract_path(account_id: @buyer.id, format: :xml, access_token: @token, id: @service_contract.id)
+      assert_response :success
+
+      xml = Nokogiri::XML::Document.parse(response.body)
+      assert xml.xpath('//service-contract/id').text == @service_contract.id.to_s
+    end
+
+    def test_success_subscribe
+      post admin_api_account_service_contracts_path(
+        account_id: @buyer.id,
+        format: :xml,
+        access_token: @token,
+        service_contract: { plan_id: @new_service_plan.id }
+      )
+      assert_response :success
+    end
+
+    def test_already_subscribed
+      post admin_api_account_service_contracts_path(
+        account_id: @buyer.id,
+        format: :xml,
+        access_token: @token,
+        service_contract: { plan_id: @service_plan.id }
```

Comment:
I think it would be better to use a different service plan (under the same service)